src/solvers/dg_simplices/analysis.jl

```diff
@@ -9,19 +8,17 @@ function calc_error_norms(func, u::StructArray, t, analyzer,
   @unpack u_values = cache
 
   # interpolate u to quadrature points
-  StructArrays.foreachfield(mul_by!(rd.Vq), u_values, u) 
-
-  # convert md.xyz::NTuple{Dim,Matrix} to StructArray for broadcasting
-  xyzq = StructArray{SVector{Dim,real(dg)}}(md.xyzq)
-  u_exact_values = initial_condition.(xyzq, t, equations) # todo: move to cache
-
-  # `pointwise_error` is a StructArray{SVector{nvariables(equations),real(dg)}}, so to square each entry 
-  # we need to apply a double broadcast (x->x.^2). 
-  pointwise_error = func.(u_values, equations) - func.(u_exact_values, equations) # todo: for loop to avoid allocations
-  component_l2_errors = sqrt.(sum(md.wJq .* (x->x.^2).(pointwise_error)))
-  component_linf_errors = maximum((x->abs.(x)).(pointwise_error))
+  StructArrays.foreachfield(mul_by!(rd.Vq), u_values, u)   
 
-  return component_l2_errors, component_linf_errors
+  component_l2_errors = zero(eltype(u_values))
+  component_linf_errors = similar(component_l2_errors)
```

Comment:
When using StaticArrays, `component_l2_errors` will be an `SVector` but `component_linf_errors` will be an `MVector - is that the desired behavior?

Reply:
That is not intended - I didn't realize that it would return `MVector`. Thanks for catching it!